Gui/opensim/view/src/org/opensim/threejs/ModelVisualizationJson.java

```diff
@@ -1344,7 +1346,8 @@ public void propagateGeometryPathCommandsToPathPoints(Muscle muscle, AbstractPro
         }
         else {
             JSONObject pathpointCommand = (JSONObject) lastCommand.clone();
-            GeometryPath gPath = muscle.getGeometryPath();
+            
+            GeometryPath gPath = GeometryPath.safeDownCast(force.getPropertyByName("GeometryPath").getValueAsObject());
```

Comment:
I would prefer avoiding `getPropertyByName()`. How do you know for sure here that the property exists? Should there be an if-statement?

Reply:
This is tricky because the presence of a GeometryPath in a Component is not built into the class hierarchy (so we can downcast and get/updGeometryPath). There's a method to check if a Force has one (hasGeometryPath) but no generic way to get it if the result is true. As it stands, the GUI calls this code only on Forces that have been checked to have GeometryPath by construction but I'm open to suggestions.

Reply:
Thanks for explaining. There is no guarantee that the force contains a property named GeometryPath. So there should be an error check.

A more maintainable solution would be to add a virtual `Force::getGeometryPath()`.

Reply:
For now, if you add an exception check and create an issue for `Force::getGeometryPath()`  I'm willing to approve the issue.

Reply:
Test and exception added.Thanks @chrisdembia

Reply:
issue https://github.com/opensim-org/opensim-core/issues/2587 has been opened to address the getPropertyByName concern,

Reply:
pinging @chrisdembia for review